zeroconf.py

```diff
@@ -1705,9 +1706,41 @@ def find(cls, zc=None, timeout=5, interfaces=InterfaceChoice.All):
         return tuple(sorted(listener.found_services))
 
 
+if "ANDROID_ARGUMENT" in os.environ:
+    from jnius import autoclass
+
+    AndroidString = autoclass("java.lang.String")
+
+    def get_network_interfaces():
+        interfaces = []
+        try:
+            NetworkInterface = autoclass("java.net.NetworkInterface")
+            Inet4Address = autoclass("java.net.Inet4Address")
+            ifaces = NetworkInterface.getNetworkInterfaces()
+            while ifaces and ifaces.hasMoreElements():
+                iface = ifaces.nextElement()
+                ips = iface.getInetAddresses()
+                while ips and ips.hasMoreElements():
+                    ip = ips.nextElement()
+                    # There are no methods for checking IP address type, so check object type.
+                    address = ip.getHostAddress()
+                    if isinstance(ip, Inet4Address):
+                        interfaces.append({'inet': address})
+                    else:
+                        interfaces.append({'inet6': address})
```

Comment:
Since we don't use the inet6 addresses here (and I think the `if` below will get a `KeyError` if it doesn't have an `"inet"` key), maybe we just skip this case.